databroker/experimental/server_ext.py

```diff
@@ -0,0 +1,501 @@
+import base64
+import collections.abc
+import json
+import os
+import uuid
+from pathlib import Path
+from typing import Dict, List
+
+import dask.array
+import h5py
+import numpy
+import pandas
+import pydantic
+import pymongo
+
+from fastapi import APIRouter, HTTPException, Request, Security
+from tiled.adapters.array import ArrayAdapter
+from tiled.adapters.dataframe import DataFrameAdapter
+from tiled.adapters.utils import IndexersMixin, tree_repr
+from tiled.server.core import json_or_msgpack
+from tiled.server.dependencies import entry
+from tiled.query_registration import QueryTranslationRegistry
+
+from tiled.structures.core import StructureFamily
+from tiled.structures.dataframe import deserialize_arrow, serialize_arrow
+
+from tiled.server.pydantic_array import ArrayStructure
+from tiled.server.pydantic_dataframe import DataFrameStructure
+from tiled.utils import APACHE_ARROW_FILE_MIME_TYPE, UNCHANGED
+
+from typing import Union
+
+from .schemas import Document
+from .queries import RawMongo
+
+from sys import platform
+
+
+class PostMetadataRequest(pydantic.BaseModel):
+    structure_family: StructureFamily
+    structure: Union[ArrayStructure, DataFrameStructure]
```

Comment:
my suggestion for fixing [this](https://github.com/bluesky/databroker/pull/696#issuecomment-1124181963) is to replace this type annotation with `Structure` from `tiled.server.schemas` which treats `micro` and `macro` as opaque dictionaries. This is the approach used in e.g. the `/node/metadata` route.

Reply:
The opaque dictionaries are a bit of cop-out / short-term corner-cutting on my part. The OpenAPI schema will be a lot more useful if we fill these out.

The specific issue here (in the linked traceback) is that OpenAPI has no idea what to do with a `pandas.DataFrame`. Seems likely we should change that to use Arrow-encoded bytes, as discussed on Slack.